project/MimaExcludes.scala

```diff
@@ -36,9 +36,11 @@ object MimaExcludes {
 
   // Exclude rules for 3.0.x
   lazy val v30excludes = v24excludes ++ Seq(
-    ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.io.SnappyCompressionCodec.version"),
+    // [SPARK-25362][JavaAPI] Replace Spark Optional class with Java Optional
+    ProblemFilters.exclude[MissingClassProblem]("org.apache.spark.api.java.Optional")
     ProblemFilters.exclude[IncompatibleMethTypeProblem]("org.apache.spark.api.java.JavaPairRDD.flatMapValues"),
```

Comment:
I realize that I neglected to leave comments in the lines above about the source of these exclusions in earlier changes. While you're here, could you add above this line: 

`// [SPARK-19287] JavaPairRDD flatMapValues requires function returning Iterable, not Iterator`